scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py

```diff
@@ -94,18 +48,22 @@ def __init__(self, parent=None, window_flags=None):
             detector.on_checkbox_checked(self.add_plot)
             detector.on_checkbox_unchecked(self.del_plot)
 
-        # peaks boxes
-        self.peaks = PeaksPresenter(PeaksView())
-        self.peaks.major.setChecked(True)
-        self.peaks.major.on_checkbox_checked(self.major_peaks_changed)
-        self.peaks.major.on_checkbox_unchecked(self.major_peaks_changed)
-        self.peaks.minor.on_checkbox_checked(self.minor_peaks_changed)
-        self.peaks.minor.on_checkbox_unchecked(self.minor_peaks_changed)
-        self.peaks.gamma.on_checkbox_checked(self.gammas_changed)
-        self.peaks.gamma.on_checkbox_unchecked(self.gammas_changed)
-        self.peaks.electron.on_checkbox_checked(self.electrons_changed)
-        self.peaks.electron.on_checkbox_unchecked(self.electrons_changed)
-        self.peaks.set_deselect_elements_slot(self.deselect_elements)
+        # plotting
+        self._plot_window = None
+        self._plotting = None
+
+        # setup periodic table and peaks
+        self.ptable_view = PeriodicTableView()
+        self.peaks_view = PeaksView()
+        self.ptable = PeriodicTableWidgetPresenter(self)
```

Comment:
why has the presenter not got a view?

Reply:
Due to structure of ElementalAnalysisGUI widget i could not seperate all the widgets relating to the periodic table into one widget

Reply:
That is the point of this piece of work. To make the periodic table a widget that obeys MVP.